File: design-patterns/delegation/csharp/Delegator.cs
using System;

delegate void DelegateFunction();

class Delegator
{
    private DelegateFunction _function;

    public void F() { _function(); }

    public void ToA() { _function = A; }
    public void ToB() { _function = B; }

    private void A() 
    { 
        Console.WriteLine("Delegate A called");
    }
    
    private void B() 
    { 
        Console.WriteLine("Delegate B called");
    }
}

File: com/growingwiththeweb/designpatterns/delegation/Delegator.java
package com.growingwiththeweb.designpatterns.delegation;

class Delegator implements DelegationInterface {
    private DelegationInterface delegate;

    public void f() { delegate.f(); }
    public void toA() { delegate = new DelegateA(); }
    public void toB() { delegate = new DelegateB(); }
}
